Feature: Added node eslint cache

SebastiaanVanspauwen opened the pull request:
- Added cache option for linting step, will use cache of current commit, if not found it will try `eslint-main` cache key
- Added optional cache eslint on merge with main, creates `eslint-main` cache key.  (added it but not sure if I am still a fan of it, does not really 'win' time since we have to run the same action twice?)

Feel free to give feedback!
Docs for ref: https://eslint.org/docs/latest/use/command-line-interface#--cache-strategy

djbe reviewed (changes requested):
1. Yeah I don't see the point of the separate prep. cache workflow… We should get rid of it. 
2. Your cache key is wrong, it'll **never** hit if you use the commit hash. It is also invalid across different runners. This is why you usually have something like `${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}` as key: differentiate by OS, specify the "type" and then key it by lock file/content.
3. Your restore key should just be a prefix (or list of prefixes) to try and match another existing cache, so we can try to use that.

So TL;DR:
```yaml
key: ${{ runner.os }}-eslint-${{ hashFiles('.eslintrc.js') }}
restore-keys: ${{ runner.os }}-eslint-
```

Note: that hashFiles could be different, just basing myself on https://github.com/eslint/eslint/discussions/16997

SebastiaanVanspauwen commented:
I removed the cache step and added the `hashFiles`
I included `eslint.config.js` for newer projects (from current nestjs template) and `.eslintrc.json` for older projects

(fyi: current setup will only hit eslint cache on second run per branch)
